src/languages/arcade.js

```diff
@@ -223,14 +246,14 @@ export default function(hljs) {
       "Variance",
       "Week",
       "Weekday",
-      "When",
+      "When|0",
       "Within",
-      "Year"
+      "Year|0",
     ]
   };
   const SYMBOL = {
     className: 'symbol',
-    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view]+'
+    begin: '\\$[datastore|feature|layer|map|measure|sourcefeature|sourcelayer|targetfeature|targetlayer|value|view|editcontext|originalFeature|featureSet|datapoint|reference|rowindex|record|config|sourcedatastore|targetdatastore|feedfeature|fencefeature|fencenotificationtype|locationupdate|measure|aggregatedFeatures|target|join|analytic]+'
```

Comment:
I think we want `()` here, not `[]`, do we not?  (yes, I realize it's a mistake that predates your PR).   So we need to fix at least that.

Optimally we'd first declared the list as an array (one item per line) and then use `regex.either` to pull it together in regex form... I'll let you decide if you want to complete that work - you can find examples in other grammars.